tests/unit/r2i/onnxrt/engine.cc

```diff
@@ -0,0 +1,139 @@
+
+class MockModel : public r2i::IModel {
+  r2i::RuntimeError Start (const std::string &name) override {r2i::RuntimeError error; return error;}
+};
+
+TEST_GROUP (OnnxrtEngine) {
+  r2i::onnxrt::Engine engine;
+  std::shared_ptr<r2i::IModel> model;
+  std::shared_ptr<r2i::IModel> inc_model;
+
+  void setup () {
+    model = std::make_shared<r2i::onnxrt::Model> ();
+    inc_model = std::make_shared<MockModel> ();
+  }
+};
+
+TEST (OnnxrtEngine, SetModel) {
+  r2i::RuntimeError error;
+
+  error = engine.SetModel (model);
+  LONGS_EQUAL (r2i::RuntimeError::Code::EOK, error.GetCode ());
+}
+
+TEST (OnnxrtEngine, SetModelNull) {
+  r2i::RuntimeError error;
+
+  error = engine.SetModel (nullptr);
+  LONGS_EQUAL (r2i::RuntimeError::Code::NULL_PARAMETER, error.GetCode ());
+}
+TEST (OnnxrtEngine, SetModelInvalid) {
+  r2i::RuntimeError error;
+
+  error = engine.SetModel (inc_model);
+  LONGS_EQUAL (r2i::RuntimeError::Code::FRAMEWORK_ERROR, error.GetCode ());
+}
+
+TEST (OnnxrtEngine, StartEngine) {
+  r2i::RuntimeError error;
+
+  error = engine.SetModel (model);
+  LONGS_EQUAL (r2i::RuntimeError::Code::EOK, error.GetCode ());
+
+  error = engine.Start ();
+  LONGS_EQUAL (r2i::RuntimeError::Code::EOK, error.GetCode ());
+}
+
+TEST (OnnxrtEngine, StartEngineEmpty) {
+  r2i::RuntimeError error;
+
+  error = engine.Start ();
+  LONGS_EQUAL (r2i::RuntimeError::Code::NULL_PARAMETER, error.GetCode ());
+}
+
+TEST (OnnxrtEngine, StartEngineTwice) {
+  r2i::RuntimeError error;
+
+  error = engine.SetModel (model);
+  LONGS_EQUAL (r2i::RuntimeError::Code::EOK, error.GetCode ());
+
+  error = engine.Start ();
+  LONGS_EQUAL (r2i::RuntimeError::Code::EOK, error.GetCode ());
+
+  error = engine.Start ();
+  LONGS_EQUAL (r2i::RuntimeError::Code::WRONG_ENGINE_STATE, error.GetCode ());
+}
+
+TEST (OnnxrtEngine, StartStopEngine) {
+  r2i::RuntimeError error;
+
+  error = engine.SetModel (model);
+  LONGS_EQUAL (r2i::RuntimeError::Code::EOK, error.GetCode ());
+
+  error = engine.Start ();
+  error = engine.Stop ();
+  LONGS_EQUAL (r2i::RuntimeError::Code::EOK, error.GetCode ());
+}
+
+TEST (OnnxrtEngine, StartStopEngineTwice) {
+  r2i::RuntimeError error;
+
+  error = engine.SetModel (model);
+  LONGS_EQUAL (r2i::RuntimeError::Code::EOK, error.GetCode ());
+
+  error = engine.Start ();
+  LONGS_EQUAL (r2i::RuntimeError::Code::EOK, error.GetCode ());
+  error = engine.Stop ();
+  LONGS_EQUAL (r2i::RuntimeError::Code::EOK, error.GetCode ());
+
+  error = engine.Start ();
+  LONGS_EQUAL (r2i::RuntimeError::Code::EOK, error.GetCode ());
+  error = engine.Stop ();
+  LONGS_EQUAL (r2i::RuntimeError::Code::EOK, error.GetCode ());
+}
+
+TEST (OnnxrtEngine, StopEngine) {
+  r2i::RuntimeError error;
+
+  error = engine.SetModel (model);
+  LONGS_EQUAL (r2i::RuntimeError::Code::EOK, error.GetCode ());
+
+  error = engine.Stop ();
+  LONGS_EQUAL (r2i::RuntimeError::Code::WRONG_ENGINE_STATE, error.GetCode ());
+}
+
+TEST (OnnxrtEngine, StopStopEngine) {
```

Comment:
There are a lot of tests missing? Where do you predict? Where do you validate that you are handling exceptions properly?